extra_playbooks/migrate_openstack_provider.yml

```diff
@@ -0,0 +1,28 @@
+---
+- hosts: kube-node:kube-master
+  tasks:
+    - name: Remove old cloud provider config
+      file:
+        path: "{{ item }}"
+        state: absent
+      with_items:
+        - /etc/kubernetes/cloud_config
+- hosts: kube-master[0]
+  tasks:
+    - name: Include kubespray-default variables
+      include_vars: ../roles/kubespray-defaults/defaults/main.yaml
+    - name: Copy get_cinder_pvs.sh to master
+      copy:
+        src: get_cinder_pvs.sh
+        dest: /tmp
+        mode: u+rwx
+    - name: Get PVs provisioned by in-tree cloud provider
+      command: /tmp/get_cinder_pvs.sh
+      register: pvs
+    - name: Remove get_cinder_pvs.sh
+      file:
+        path: /tmp/get_cinder_pvs.sh
+        state: absent
+    - name: Rewrite the "pv.kubernetes.io/provisioned-by" annotation
+      command: "{{ bin_dir }}/kubectl annotate --overwrite pv {{ item }} pv.kubernetes.io/provisioned-by=cinder.csi.openstack.org"
```

Comment:
I was wondering, is this enough to allow the CSI driver to delete the cinder volume that was provisioned by the in-tree provisioner from OpenStack once the PVC is removed?
The CSIMigration feature gate doesn't help with such things? I was reading https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-migration.md and I am not sure how everything is implemented with the feature gate.

Reply:
Correct this is everything needed for the volumes to be deleted.

The official upgrade instructions can be found here: https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/migrate-to-ccm-with-csimigration.md#caveats

The feature might be fixed in the future however for all current kubernetes releases this workaround need to be applied

Reply:
Great! Thanks for the contribution :D

Reply:
/lgtm